File: client/src/app/models/order.ts
export interface Order{
    id: number;
    buyerEmail: string;
    orderDate: string;
    shippingAddress: ShippingAddress;
    deliveryFee: number;
    subtotal: number;
    total: number;
    discount: number;
    orderStatus: string;
    orderItems: OrderItem[];
    paymentSummary: PaymentSummary;
}

export interface ShippingAddress{
    name: string;
    line1: string;
    line2?: string | null;
    city: string;
    state: string;
    postal_code: string;
    country: string;
}

export interface OrderItem{
    productId: number;
    name: string;
    pictureUrl: string;
    price: number;
    quantity: number;
}

export interface PaymentSummary{
    last4: number | string;
    brand: string;
    exp_month: number;
    exp_year: number;
}

export interface CreateOrder{
    shippingAddress: ShippingAddress;
    paymentSummary: PaymentSummary;
}

File: API/Entities/OrderAggregrate/Order.cs
using System;

namespace API.Entities.OrderAggregrate;

public class Order
{
    public int Id { get; set; }
    public required string BuyerEmail { get; set; }
    public DateTime OrderDate { get; set; } = DateTime.UtcNow;
    public required ShippingAddress ShippingAddress { get; set; }
    public List<OrderItem> OrderItems { get; set; } = [];
    public long Subtotal { get; set; }
    public long DeliveryFee { get; set; }
    public long Discount { get; set; }
    public required string PaymentIntentId { get; set; }
    public PaymentSummary? PaymentSummary { get; set; }
    public OrderStatus Status { get; set; } = OrderStatus.Pending;

    public long GetTotal()
    {
        return Subtotal + DeliveryFee - Discount;
    }

    public static implicit operator Task<object>(Order v)
    {
        throw new NotImplementedException();
    }
}
